src/components/manager.ts

```diff
@@ -840,11 +841,20 @@ export class Manager {
         }
     }
 
+    // This function toggles the autobuild state and returns whether autobuild is enabled.
+    toggleAutoBuild() {
+        this.disableAutoBuild = !this.disableAutoBuild
+        return !this.disableAutoBuild
```

Comment:
I think having two settings to control the same feature is a bad idea.
You should rather change the value of `latex.autoBuild.run` directly.

Reply:
Agreed. But there are two problems that hinder the use of `latex.autoBuild.run`. 1) Currently `latex.autoBuild.run` is a string and `onFileChange` is the only option that is meaningful. But I don't know whether other new options will be added to `latex.audoBuild.run` and conflict with the auto build toggle. 2) `latex.autoBuild.run` is a user configuration retrieved by `vscode.workspace.getConfiguration`. It seems that it's not possible to change user configurations via code (at least I didn't get an approach when I googled last few days).

Reply:
For 2), you could use the `update` method, see https://code.visualstudio.com/api/references/vscode-api#WorkspaceConfiguration

For 1), you have to assume that the current two values will not be changed.
The proper way to handle this would be turn `latex-workshop.latex.autoBuild.run` into a boolean `latex-workshop.latex.autoBuild.enabled`. Let me think about it.

Reply:
As suggested in #2209, you could define

```ts
const enum BuildEvents {
    never = 'never',
    onFileChange = 'onFileChange'
}
```
Let me know when your code is refactored.

Reply:
Just finished my refactoring.
By the way, I rebased the commits to the latest master branch and chose to use the clock icon in the sidebar.
😄